Makefile

```diff
@@ -322,7 +323,11 @@ test/addons-napi/.buildstamp: config.gypi \
 	src/node_api.h src/node_api_types.h
 #	Cannot use $(wildcard test/addons-napi/*/) here, it's evaluated before
 #	embedded addons have been generated from the documentation.
+#	Ignore folders without binding.gyp
+#	(https://github.com/nodejs/node/issues/14843)
 	@for dirname in test/addons-napi/*/; do \
+		if [ ! -f "$$PWD/$${dirname}binding.gyp" ]; then \
+			continue; fi ; \
```

Comment:
Suggestion: it might be cleaner if you change

```bash
	if [ ! -f "$$PWD/$${dirname}binding.gyp" ]; then \
			continue; fi ; \
```

to

```bash
[ ! -f "$$PWD/$${dirname}binding.gyp" ] && continue ; \
```

Maybe it's just me that find that cleaner though.

Reply:
I think moving control flow into logical expressions is … hm. :smile:

Reply:
It just wasn't immediately obvious where the end of the `if` block was, because the `fi` isn't on its own line.

As this already squashes things down, you could go further and do:

```bash
		if [ ! -f "$$PWD/$${dirname}binding.gyp" ]; then continue; fi ; \
```

>I think moving control flow into logical expressions is … hm. 😄

Unrelated to this PR, but I'd be interested to know what the objection is, I've heard a lot of people say this, but for me `[ x ] && y` is just a cleaner way of saying `if [ x ]; then y; fi` (for simple `y`s). Obviously in a language with nicer `if`s, then there's not much gain (e.g. for rust `if x { y }`), but in bash it seems more worthwhile.

Reply:
@gibfahn I really like `[ x ] && y` too, but for other things – shell scripts are exceptional in that you can actually make control flow part of the expression, and that just feels pretty odd to me I guess?